docs/modin_xgboost.rst

```diff
@@ -0,0 +1,142 @@
+------------------
+
+Data is passed to ``modin.experimental.xgboost`` functions via a ``ModinDMatrix`` object.
+
+The ``ModinDMatrix`` stores data as Modin DataFrames internally. 
+
+Currently, the ``ModinDMatrix`` supports ``modin.pandas.DataFrame`` only as an input.
+
+
+A Single Node / Cluster setup
+-----------------------------
+
+The XGBoost part of Modin uses a Ray resources by similar way as all Modin functions.
+
+To start the Ray runtime on a single node:
+
+.. code-block:: python
+
+  import ray
+  ray.init()
+
+If you already had the Ray cluster you can connect to it by next way:
+
+.. code-block:: python
+
+  import ray
+  ray.init(address='auto')
+
+A detailed information about initializing the Ray runtime you can find in `starting ray`_  page.
+
+
+Usage example
+-------------
+
+In example below we train XGBoost model using `the Iris Dataset`_ and get prediction on the same data.
+All processing will be in a `single node` mode.
+
+.. code-block:: python
+
+  from sklearn import datasets
+  
+  import ray
+  ray.init() # Start the Ray runtime for single-node
+  
+  import modin.pandas as pd
+  import modin.experimental.xgboost as xgb
+  
+  # Load iris dataset from sklearn
+  iris = datasets.load_iris()
+  
+  # Create Modin DataFrames
+  X = pd.DataFrame(iris.data)
+  y = pd.DataFrame(iris.target)
+  
+  # Create ModinDMatrix
+  dtrain = xgb.ModinDMatrix(X, y)
+  dtest = xgb.ModinDMatrix(X, y)
+  
+  # Set training parameters
+  xgb_params = {
+      "eta": 0.3,
+      "max_depth": 3,
+      "objective": "multi:softprob",
+      "num_class": 3,
+      "eval_metric": "mlogloss",
+  }
+  steps = 20
+  
+  # Run training
+  model = xgb.train(
+      xgb_params,
+      dtrain,
+      steps,
+      evals=[(dtrain, "train")]
+  )
+  
+  # Save for some usage
+  evals_result = model["history"]
+  booster = model["booster"]
+  
+  # Predict results
+  prediction = xgb.predict(model, dtest)
+
+
+Modes of a data distribution
+----------------------------
+
+Modin XGBoost provides two approaches for an internal data ditribution which could be
+switched by `evenly_data_distribution` parameter of ``train/predict`` functions:
+
+* `evenly_data_distribution` = `True`: in this case the input data of ``train/predict``
+  functions will be distributed evenly between nodes in a cluster to ensure evenly utilization of nodes (default behavior).
+
+* `evenly_data_distribution` = `False` :  in this case partitions of input data of ``train/predict``
+  functions will not transfer between nodes in cluster in case empty nodes is <10%,
+  if portion of empty nodes is ≥10% evenly data distribution will be applied.
+  This method provides minimal data transfers between nodes but doesn't guarantee effective utilization of nodes.
+  Most effective in case when all cluster nodes are occupied by data.
+
+
+.. _Dataframe: https://pandas.pydata.org/pandas-docs/version/0.23.4/generated/pandas.DataFrame.html
+.. _`installation page`: installation.html
```

Comment:
.html -> .rst because the links are incorrect. Also, if there are some more places with them, can you fix those in a separate issue?

Reply:
Why is it needed? In all places `*.html` is used and successfully worked.

Reply:
If you use full path (https://modin.readthedocs.io/en/latest/installation.html, for example), it works. But if you use just `installation.html`, it says `This is not a web page you are looking fore`.

Reply:
Do you use `sphinx ` for compile webpages?

Reply:
@YarShev All links was changed using `:doc: name <path>` sphinx template. Absolute address links don't used due to
possible inconsistency versions of referenced doc files from repo and from https://modin.readthedocs.io/. And as I viewed most of open-source projects make internal links by this way. This links don't work when browsing using github repo, but works in a separate web-site (the similar behavior we can see for xgboost, numpy, ray etc).

Reply:
Please, fix it as a part of another issue. That doesn't belong the changes of this PR.

Reply:
This changes was moved to #2654